File: WasteVisionWebBE/Domain/Roles/RoleMapper.cs
using System.Collections.Generic;
using System.Linq;
using DDDSample1.Domain.Shared;

namespace DDDSample1.Domain.Roles
{
    public static class RoleMapper
    {
        public static RoleDto ToDto(Role role)
        {
            if (role == null)
                return null;

            return new RoleDto
            {
                Id = role.Id.AsGuid(),
                Description = role.Description.AsString(),
                Active = role.Active
            };
        }

        public static Role ToDomain(CreatingRoleDto dto)
        {
            if (dto == null)
                return null;

            return new Role(
                new Description(dto.Description)
            );
        }

        public static List<RoleDto> ToDtoList(List<Role> roles)
        {
            if (roles == null)
                return null;

            return roles.Select(role => ToDto(role)).ToList();
        }

        public static void UpdateDomain(Role role, RoleDto dto)
        {
            if (role == null || dto == null)
                return;

            if (!string.IsNullOrWhiteSpace(dto.Description))
                role.ChangeDescription(new Description(dto.Description));
        }
    }
}

File: WasteVisionWebFE/src/data/mappers/role-mapper.ts
import { Role } from "@/core/domain/Role";
import {
  RoleDTO,
  CreateRoleDTO,
  UpdateRoleDTO,
  RoleSearchParamsDTO,
} from "@/data/dto/role-dto";

export class RoleMapper {
  /**
   * Maps a RoleDTO to a domain Role
   */
  public static toDomain(dto: RoleDTO): Role {
    return {
      id: dto.id,
      description: dto.description,
      active: dto.active,
    };
  }

  /**
   * Maps a domain Role to a RoleDTO
   */
  public static toDTO(domain: Role): RoleDTO {
    return {
      id: domain.id,
      description: domain.description,
      active: domain.active,
    };
  }

  /**
   * Maps an array of RoleDTOs to domain Roles
   */
  public static toDomainList(dtos: RoleDTO[]): Role[] {
    return dtos.map((dto) => this.toDomain(dto));
  }

  /**
   * Maps a domain Role to a CreateRoleDTO
   */
  public static toCreateDTO(domain: Role): CreateRoleDTO {
    return {
      description: domain.description,
    };
  }

  /**
   * Maps a domain Role to an UpdateRoleDTO
   */
  public static toUpdateDTO(domain: Role): UpdateRoleDTO {
    return {
      description: domain.description,
    };
  }

  /**
   * Maps search parameters to a RoleSearchParamsDTO
   */
  public static toSearchParamsDTO(description?: string): RoleSearchParamsDTO {
    return {
      description,
    };
  }
}
